packages/fleather/lib/src/widgets/controller.dart

```diff
@@ -69,14 +70,40 @@ class FleatherController extends ChangeNotifier {
         .mergeAll(toggledStyles);
   }
 
-  bool _shouldApplyToggledStyles(Delta delta) =>
-      toggledStyles.isNotEmpty &&
-      delta.isNotEmpty &&
-      ((delta.length <= 2 && // covers single insert and a retain+insert
-              delta.last.isInsert) ||
-          (delta.length <= 3 &&
-              delta.last.isRetain // special case for AutoTextDirectionRule
-          ));
+  bool _shouldApplyToggledStyles(Delta delta) {
+    if (toggledStyles.isNotEmpty && delta.isNotEmpty) {
+      // covers single insert and a retain+insert
+      if (delta.length <= 2 && delta.last.isInsert) {
+        return true;
+      }
+      // special case for AutoTextDirectionRule
+      if (delta.length <= 3 && delta.last.isRetain) {
+        return delta.last.attributes
```

Comment:
Wouldn't it be easier to read like this?
```dart
return delta.last.attributes != null && delta.last.attributes!.contains... && ....;
```

WDYT?

Reply:
Agreed.